service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt

```diff
@@ -370,153 +375,159 @@ fun addServiceNeedDataToVarda(db: Database.Connection, vardaClient: VardaClient,
+                vardaChildId = newVardaServiceNeed.vardaChildId!!,
+                evakaServiceNeedInfoForVarda = evakaServiceNeed,
+                guardians = guardians
+            ) ?: error("VardaUpdate: client failed voucher request for $feeId - see logs for details")
+        } catch (e: Exception) {
+            error { "VardaUpdate: failed to send voucher decision data for service need ${evakaServiceNeed.id}: ${e.localizedMessage}" }
+        }
+    }
 
-fun getVardaGuardian(client: VardaClient, ssn: String): VardaGuardian {
-    val person = client.getPersonFromVardaBySSN(ssn) ?: error("VardaUpdate: Couldn't fetch guardian from Varda")
-    return VardaGuardian(
-        henkilotunnus = ssn,
-        etunimet = person.firstName,
-        sukunimi = person.lastName
-    )
+    return feeResponseIds + voucherResponseIds
 }
 
 data class VardaGuardianWithId(
     val id: UUID,
-    val henkilotunnus: String,
+    val henkilotunnus: String?,
+    val henkilo_oid: String?,
     val etunimet: String,
     val sukunimi: String
 ) {
     fun toVardaGuardian(): VardaGuardian = VardaGuardian(
         henkilotunnus = henkilotunnus,
+        henkilo_oid = henkilo_oid,
         etunimet = etunimet,
         sukunimi = sukunimi
     )
 }
 
 fun getChildVardaGuardians(db: Database.Connection, childId: UUID): List<VardaGuardianWithId> {
     return db.read {
-        it.createQuery("select id, first_name AS etunimet, last_name as sukunimi, social_security_number AS henkilotunnus FROM person where id IN (SELECT guardian_id FROM guardian WHERE child_id = :id)")
+        it.createQuery("select id, first_name AS etunimet, last_name as sukunimi, social_security_number AS henkilotunnus, oph_person_oid AS henkilo_oid FROM person where id IN (SELECT guardian_id FROM guardian WHERE child_id = :id)")
             .bind("id", childId)
             .mapTo<VardaGuardianWithId>()
             .list()
     }
 }
 
-fun createFeeDataToVardaFromVoucherValueDecision(
-    db: Database.Connection,
+fun getGuardianFromVarda(client: VardaClient, ssn: String?, oid: String?): VardaGuardian {
+    val person = client.getPersonFromVardaBySsnOrOid(ssn, oid) ?: error("VardaUpdate: Couldn't fetch guardian from Varda")
+    return VardaGuardian(
+        henkilotunnus = ssn,
+        henkilo_oid = person.personOid,
+        etunimet = person.firstName,
+        sukunimi = person.lastName
+    )
+}
+
+fun getFeeReceiverGuardians(db: Database.Connection, childId: UUID, headOfFamilyId: UUID): List<VardaGuardianWithId> {
+    val guardians = getChildVardaGuardians(db, childId)
+    if (guardians.isEmpty()) error("VardaUpdate: could not create fee data: child has no guardians")
+
+    // If head of family is not a guardian, fee data is not supposed to be sent to varda (https://wiki.eduuni.fi/display/OPHPALV/Huoltajan+tiedot)
+    if (guardians.none { it.id == headOfFamilyId }) {
+        error("VardaUpdate: could not create fee data: head of family is not a guardian")
```

Comment:
Is this actually an error? I mean I understand the control flow here pretty much currently requires it to be but purely from a business logic POV -- isn't this just data that shouldn't be sent (maybe worth info-level logging, though)? Or am I misunderstanding the linked docs?

Maybe what I'm really asking is (assuming this shouldn't be an actual error): should these cases be filtered out earlier in the flow?

Reply:
That's a good question. I think we would want to get notified on these at least on some level. But on the other hand, apart from perhaps a bunch of cases when these are indications of bad data, logging these as exceptions is probably just noise.

Reply:
I'm OK with keeping these as errors for now (assuming there's no other clear solution) and just keeping it in mind, and maybe later removing it once all data is in a more acceptable condition.

Reply:
But considering that the data is handled childwise I think we should do the filtering in a very early step, before the actual processing of service needs.

Reply:
Also if we did that, then this exception would be justified because each processed child should have a guardian as their head of family

Reply:
Agreed to do the pre-filtering later and merge this as-is